### PR TITLE
New Obstacle: Brambles(want to be merged)

### PR DESCRIPTION
**Pull request by RYQ-22**
荆棘类型的Obstacle，子弹和unit可以穿透，对所有在其上的unit持续降低生命值

### CODE REVIEW

**Comment by RYQ-22** [comment]
Brambles

**Comment by LazyJazz** [comment]
Seems like you are doing the very same thing with #204

**Comment by RYQ-22** [comment]
> > Seems like you are doing the very same thing with #204
> 
> My implementation can support all units, not just tiny_tank

I did not modify the code of the units part.

File: CMakeSettings.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      "configurationType": "Debug",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "buildRoot": "${projectDir}\\out\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "",
+      "ctestCommandArgs": ""
+    }
+  ]
+}
```
Comment (LazyJazz) [inline CMakeSettings.json:15]: Delete this file. add it to .gitginore

**Review by LazyJazz — CHANGES_REQUESTED** [review]
Nice job, just fix the nits

File: CMakeSettings.json (hunk 1 above)
Comment (RYQ-22) [reply CMakeSettings.json]: Has been modified.

File: .gitignore [hunk 2]
```diff
@@ -6,4 +6,19 @@ build
...
+      "configurationType": "Debug",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "buildRoot": "${projectDir}\\out\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "",
+      "ctestCommandArgs": ""
+    }
+  ]
+}
```
Comment (LazyJazz) [inline .gitignore:24]: I mean, add the filename into this file. which could prevent add it again.

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (RYQ-22) [reply .gitignore]: Has been modified.

**Comment by qinchuanhui** [comment]
Plz resolve the conflicts

**Review by qinchuanhui — APPROVED** [review]
LGTM, you can get the corresponding score.
But don't merge directly due to the format error

**Pull request closed without merging**